### PR TITLE
cmd/faucet: switch Facebook auth over to mobile site

### PR DESCRIPTION
**Pull request by karalabe**
The full website uses AJAX. Switch over to the mobile site for scraping the infos.

### CODE REVIEW

File: cmd/faucet/faucet.go [hunk 1]
```diff
@@ -847,7 +847,13 @@ func authFacebook(url string) (string, string, common.Address, error) {
 	// Facebook's Graph API isn't really friendly with direct links. Still, we don't
 	// want to do ask read permissions from users, so just load the public posts and
 	// scrape it for the Ethereum address and profile URL.
-	res, err := http.Get(url)
+	//
+	// Facebook recently changed their desktop webpage to use AJAX for loading post
+	// content, so switch over to the mobile sit for now. Will probably end up having
```
Comment (ligi) [inline cmd/faucet/faucet.go:852]:
```suggestion
	// content, so switching over to the mobile site for now. Will probably end up having
```

File: cmd/faucet/faucet.go [hunk 2]
```diff
@@ -847,7 +847,13 @@ func authFacebook(url string) (string, string, common.Address, error) {
...
+	// to do the API eventually.
```
Comment (ligi) [inline cmd/faucet/faucet.go:853]:
```suggestion
	// to use the API eventually.
```

**Review by ligi — CHANGES_REQUESTED** [review]

File: cmd/faucet/faucet.go (hunk 1 above)
Comment (karalabe) [reply cmd/faucet/faucet.go]: Fixed

File: cmd/faucet/faucet.go (hunk 2 above)
Comment (karalabe) [reply cmd/faucet/faucet.go]: Fixed

**Comment by holiman** [comment]
How about adding `cmd/faucet/faucet_test.go`: 
```golang
package main

import (
	"fmt"
	"testing"

	"github.com/ethereum/go-ethereum/common"
)

func TestFacebook(t *testing.T) {
	for _, tt := range []struct {
		url  string
		want common.Address
	}{
		{
			"https://www.facebook.com/ambreen.fatimah.1/posts/781618749097122",
			common.HexToAddress("0x956Bcb03a6f8Df9566C7b300E35A21571e30c559"),
		},
	} {
		_, _, gotAddress, err := authFacebook(tt.url)
		if err != nil {
			t.Fatal(err)
		}
		if gotAddress != tt.want {
			t.Fatalf("address wrong, have %v want %v", gotAddress, tt.want)
		}
	}
}
```

**Comment by holiman** [comment]
( the reason I haven't pushed it as  a commit, is that it would be better to use facebook-posts that aren't just random people, but people who are already affiliated with go-ethereum. I don't have a FB account personally... Although, hold on, I do have an account or two..)

**Comment by holiman** [comment]
Ok, this one should work: 
```golang
package main

import (
	"testing"

	"github.com/ethereum/go-ethereum/common"
)

func TestFacebook(t *testing.T) {
	for _, tt := range []struct {
		url  string
		want common.Address
	}{
		{
			"https://www.facebook.com/fooz.gazonk/posts/2837228539847129",
			common.HexToAddress("0xDeadDeaDDeaDbEefbEeFbEEfBeeFBeefBeeFbEEF"),
		},
	} {
		_, _, gotAddress, err := authFacebook(tt.url)
		if err != nil {
			t.Fatal(err)
		}
		if gotAddress != tt.want {
			t.Fatalf("address wrong, have %v want %v", gotAddress, tt.want)
		}
	}
}
```

**Comment by karalabe** [comment]
@holiman PTAL

**Pull request merged**